db/core/src/test/java/org/ethereum/beacon/db/source/impl/XorDataSourceTest.java

```diff
@@ -0,0 +1,16 @@
+package org.ethereum.beacon.db.source.impl;
+
+import org.junit.Test;
+import tech.pegasys.artemis.util.bytes.BytesValue;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class XorDataSourceTest {
+
+    @Test
+    public void testValidSourceCreation() {
+        assertThatThrownBy(() -> new XorDataSource<>(null, BytesValue.EMPTY))
+                .isInstanceOf(NullPointerException.class);
+    }
+
```

Comment:
XorDataSource<>(someDatasource, null) case is missing
No tests for XorDataSource functionality (private static BytesValue xorLongest(BytesValue v1, BytesValue v2) method)